finetune/adapter.py

```diff
@@ -229,18 +233,12 @@ def validate(
     output = tokenizer.decode(output)
     fabric.print(output)
 
+    model.reset_cache()
```

Comment:
This saves memory but very little, as we pass `max_seq_length=max_returned_tokens` above, so it's not using the entire `max_seq_length`.